zingolib/src/wallet/transaction_record.rs

```diff
@@ -0,0 +1,343 @@
+use incrementalmerkletree::witness::IncrementalWitness;
+use zcash_primitives::{sapling, transaction::TxId};
+
+use crate::error::ZingoLibError;
+
+use super::{
+    data::{OutgoingTxData, PoolNullifier},
+    *,
+};
+
+///  Everything (SOMETHING) about a transaction
+#[derive(Debug)]
+pub struct TransactionRecord {
+    // the relationship of the transaction to the blockchain. can be either Broadcast (to mempool}, or Confirmed.
+    pub status: ConfirmationStatus,
+
+    // Timestamp of Tx. Added in v4
+    pub datetime: u64,
+
+    // Txid of this transaction. It's duplicated here (It is also the Key in the HashMap that points to this
+    // WalletTx in LightWallet::txs)
+    pub txid: TxId,
+
+    // List of all nullifiers spent in this Tx. These nullifiers belong to the wallet.
+    pub spent_sapling_nullifiers: Vec<zcash_primitives::sapling::Nullifier>,
+
+    // List of all nullifiers spent in this Tx. These nullifiers belong to the wallet.
+    pub spent_orchard_nullifiers: Vec<orchard::note::Nullifier>,
+
+    // List of all sapling notes received in this tx. Some of these might be change notes.
+    pub sapling_notes: Vec<SaplingNote>,
+
+    // List of all sapling notes received in this tx. Some of these might be change notes.
+    pub orchard_notes: Vec<OrchardNote>,
+
+    // List of all Utxos received in this Tx. Some of these might be change notes
+    pub transparent_notes: Vec<TransparentNote>,
+
+    // Total value of all the sapling nullifiers that were spent in this Tx
+    pub total_sapling_value_spent: u64,
+
+    // Total value of all the orchard nullifiers that were spent in this Tx
+    pub total_orchard_value_spent: u64,
+
```

Comment:
Between the names of these fields and the docs, it _looks_ like these values will always be available and reliable. But for _incoming_ transactions I think that cannot be the case. How can we set appropriate expectations there?
Maybe just change "all the (pool) nullifiers that were spent" to "all the (pool) nullifiers spent by this account".